app/battle.py

```diff
@@ -0,0 +1,17 @@
+from app.knight import Knight
+
+
+class Battle:
+    def __init__(self, knight1: Knight, knight2: Knight) -> None:
+        self.knight1 = knight1
+        self.knight2 = knight2
+
+    def conduct_battle(self) -> None:
+        self.knight1.adjust_hp(self.knight2.total_power)
+        self.knight2.adjust_hp(self.knight1.total_power)
```

Comment:
The `conduct_battle` method does not account for the knights' protection when calculating the damage. According to the task description, the damage should be calculated as `opponent_power - self_protection`. You should adjust the HP using this formula to ensure the battle logic is correct.